src/mapnik_vector_tile.cpp

```diff
@@ -1688,89 +1605,94 @@ NAN_METHOD(VectorTile::toJSON)
 {
     NanScope();
     VectorTile* d = node::ObjectWrap::Unwrap<VectorTile>(args.Holder());
-    vector_tile::Tile const& tiledata = d->get_tile();
-    Local<Array> arr = NanNew<Array>(tiledata.layers_size());
-    for (int i=0; i < tiledata.layers_size(); ++i)
-    {
-        vector_tile::Tile_Layer const& layer = tiledata.layers(i);
-        Local<Object> layer_obj = NanNew<Object>();
-        layer_obj->Set(NanNew("name"), NanNew(layer.name().c_str()));
-        layer_obj->Set(NanNew("extent"), NanNew<Integer>(layer.extent()));
-        layer_obj->Set(NanNew("version"), NanNew<Integer>(layer.version()));
-
-        Local<Array> f_arr = NanNew<Array>(layer.features_size());
-        for (int j=0; j < layer.features_size(); ++j)
-        {
-            Local<Object> feature_obj = NanNew<Object>();
-            vector_tile::Tile_Feature const& f = layer.features(j);
-            if (f.has_id())
-            {
-                feature_obj->Set(NanNew("id"),NanNew<Number>(f.id()));
-            }
-            if (f.has_raster())
-            {
-                std::string const& raster = f.raster();
-                feature_obj->Set(NanNew("raster"),NanNewBufferHandle((char*)raster.data(),raster.size()));
-            }
-            feature_obj->Set(NanNew("type"),NanNew<Integer>(f.type()));
-            Local<Array> g_arr = NanNew<Array>();
-            for (int k = 0; k < f.geometry_size();++k)
-            {
-                g_arr->Set(k,NanNew<Number>(f.geometry(k)));
-            }
-            feature_obj->Set(NanNew("geometry"),g_arr);
-            Local<Object> att_obj = NanNew<Object>();
-            for (int m = 0; m < f.tags_size(); m += 2)
+    try {
```

Comment:
The scope of this try section is rather large, any reason it is so large and could we possible reduce it in scope?